tests/utility/diff.zig

```diff
@@ -1,30 +1,37 @@
 const std = @import("std");
 const zls = @import("zls");
 
-const allocator = std.testing.allocator;
-
 fn gen(alloc: std.mem.Allocator, rand: std.rand.Random) ![]const u8 {
-    var buffer = try alloc.alloc(u8, rand.intRangeAtMost(usize, 16, 1024));
-    for (buffer) |*b| b.* = rand.intRangeAtMost(u8, 32, 126);
+    var buffer = try alloc.alloc(u8, rand.intRangeAtMost(usize, 0, 256));
+    for (buffer) |*b| b.* = rand.intRangeAtMost(u8, ' ', '~');
     return buffer;
 }
 
 test "diff - random" {
-    var arena = std.heap.ArenaAllocator.init(allocator);
-    defer arena.deinit();
-
-    var rand = std.rand.DefaultPrng.init(0);
-
-    var index: usize = 0;
+    const allocator = std.testing.allocator;
+    try std.testing.checkAllAllocationFailures(allocator, testDiff, .{ 0, .@"utf-8" });
```

Comment:
Mom, can we have borrow checker?
No, we already have borrow checker at home.
at home:
`std.testing.checkAllAllocationFailures`